sympy/combinatorics/homomorphisms.py

```diff
@@ -301,7 +301,7 @@ def homomorphism(domain, codomain, gens, images=[], check=True):
     gens = list(gens)
     images = list(images)
 
-    images.extend([codomain.identity]*(len(generators)-len(images)))
+    images.extend((codomain.identity,) * (len(generators)-len(images)))
```

Comment:
What advantage does a change like this give?

It could alternatively be:
```python
images += [codomain.identity] * (len(generators)-len(images))
```
I think that's potentially fastest and also nicer to read IMO.

Reply:
There are many other changes like this one

Reply:
Speed, I'd say. But your argument is primarily against variables not constants?

Not sure one can say that += will always be faster?

Reply:
Can you show a timing that demonstrates the speed improvement?

Reply:
Given these examples:
```python
def f1(a):
    l = []
    for _ in range(100):
        l.extend([a] * 100)

def f2(a):
    l = []
    for _ in range(100):
        l.extend((a,) * 100)

def f3(a):
    l = []
    for _ in range(100):
        l += [a] * 100
```
I get the timings
```python
In [22]: %timeit f1(5)
210 µs ± 8.98 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [23]: %timeit f2(5)
248 µs ± 2.79 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [24]: %timeit f3(5)
213 µs ± 11.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```
According to that `f2` (using a tuple) is slowest. I'm not sure how well that generalises but it doesn't support making the change here.

Ultimately I think that if we're not sure about a change it's best just to stick with what we have.

Reply:
OK! I might have generalized things a bit to far. With constants there is a huge speed-up using tuples, but not using variables.

Reply:
```
%timeit (x,)*100
621 ns ± 3.82 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

%timeit [x,]*100
578 ns ± 7.2 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

%timeit [1,]*100
562 ns ± 29.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

%timeit (1,)*100
14.6 ns ± 0.25 ns per loop (mean ± std. dev. of 7 runs, 100000000 loops each)
```

Reply:
I think that the huge speedup is only when the whole tuple can be precomputed at compile time.

Otherwise there is no reason in principle why creating a tuple should be faster than creating a list either for small or large sizes.